If manage_user=false, don't depend on the User resource

jake2184 opened the pull request:
#### Pull Request (PR) description
If ``manage_user=false``, don't depend on the existence of a User[] resource.

We are forced to rely on a user supplied by LDAP, which is not created (and cannot be edited by) a User[] resource. 

The only negative I can think of to this PR is if somebody is defining a User resource elsewhere, setting manage_user to false, and then expecting resources within this module to be subscribed to it. It seems unlikely but possible.

For reference, we are following the same methodology with BitBucket, where it behaves as expected if ``manage_usr_grp`` is set to false.

alexjfisher commented on manifests/install.pp:
```diff
@@ -63,9 +63,11 @@
           File[$confluence::webappdir] ],
       }
+      if ($confluence::manage_user) {
+        Staging::Extract[$file] -> User[$confluence::user]
```
This is the wrong way round?  Should be
`User[$confluence::user] -> Staging::Extract[$file]` ?

alexjfisher commented on manifests/install.pp:
```diff
@@ -63,9 +63,11 @@
-          User[$confluence::user],
           File[$confluence::webappdir] ],
       }
+      if ($confluence::manage_user) {
```
brackets aren't required

alexjfisher commented on manifests/install.pp:
```diff
@@ -104,6 +108,8 @@
-    subscribe   => User[$confluence::user],
+  }
+  if ($confluence::manage_user) {
+    Exec["chown_${confluence::webappdir}"] ~> User[$confluence::user]
```
User should notify the Exec, not the other way.

jake2184 commented:
oops, yes. New to the syntax, and didn't reverse the 'B requires A' into 'A -> B'. 

Have reversed, and added tests. I didn't want to do those until the premise was accepted - again, new to it so the test take me time. 

I think I've addressed all issues with an amended commit, waiting for checks to pass (they do locally, I remembered to check before pushing this time..)

kenyon commented:
Closed by #216.
